docs/get_started/single_agent.md

```diff
@@ -0,0 +1,142 @@
+- **Communication**: our framework allows flexible and scalable communication between agents. This is fundamental for the critical research question.
+- **Reasoning**: we will equip agents with different planning and reward (critic) learning abilities, allowing them to optimize task completion in a more guided approach.
+
+
+
+<!-- - (WIP) **Reasoning Ability**: since any goal can be formalized as the outcome of maximizing cumulative rewards, we will equip our agents with policy which they could follow to achieve goals. -->
+
+<!-- We will first start with the single agent setting, where the agent can interact with users, process and store messages, and utilize external tools to generate responses and accomplish tasks. -->
+
+## Quick Start
+Let's first play with a `ChatAgent` instance by simply initialize it with a system message and interact with user messages.
+
+### Step 0: Prepartions
+```python
+from camel.messages import BaseMessage as bm
+from camel.agents import ChatAgent
+```
+
+### Step 1: Define the Role
+Create a system message to define agent's default role and behaviors.
+```python
+sys_msg = bm.make_assistant_message(
+    role_name='stone',
+    content='you are a curious stone wondering about the universe.')
+```
+
+### Step 2: Initialize the Agent
+```python
+agent = ChatAgent(
+    system_message=sys_msg,
+    message_window_size=10,    # [Optional] the length for chat memory
+    )
+```
+### Step 3: Interact with the Agent with `.step()`
+```python
+# Define a user message
+usr_msg = bm.make_user_message(
+    role_name='prof. claude shannon',
+    content='what is information in your mind?')
+
+# Sending the message to the agent
+response = agent.step(usr_msg)
+
+# Check the response (just for illustrative purpose)
+print(response.msgs[0].content)
+>>> information is the resolution of uncertainty.
+```
+Woohoo, your first agent is ready to play with you!
+
+
+## Advanced Features
+
+### Tool Usage
+```python
+# Import the necessary functions
+from camel.functions import MATH_FUNCS, SEARCH_FUNCS
+
+# Initialize the agent with list of functions
+agent = ChatAgent(
+    system_message=sys_msg,        
+    function_list=[*MATH_FUNCS, *SEARCH_FUNCS]
+    )
+
+# Check if functions are enabled
+agent.is_function_calling_enabled()
+>>> True
+```
+
+### Memory
+By default our agent is initialized with `ChatHistoryMemory`, allowing agents to do in-context learning, though restricted by the finite window length.
+
+Assume that you have followed the setup in [Quick Start](#quick-start). Let's first check what is inside its brain.
+```python
+agent.memory.get_context()
+>>> ([{'role': 'system', 'content': 'you are a helpful assistant.'},
+      {'role': 'user', 'content': 'what is information in your mind?'}],
+      30)
+```
+By default, only the user messages are saved. You may update/alter the agent's memory with any externally provided message in the format of `BaseMessage`; for example, using the agent's own response:
```

Comment:
By default, I think the assistant messages will also be saved. Besides, all memory operation is encupsulated in `ChatAgent.step()`, we may not need user to record messages by themselves?

Reply:
ah, i just checked the latest release pip release, it seems that in this version only the user message is saved. how about adding a clarification on the potential change in data handling in the next release? btw when should people expect that?

Reply:
Yeah, it has been a long time since last release. I think the "only the user message is saved" is a bug rather than a "feature". I would prefer to change it now because the docs in the repo match the `master` branch version, not the release version.

Reply:
And I think the next release will be soon.

Reply:
just removed those lines